chart/templates/connection-secret-job.yaml

```diff
@@ -0,0 +1,96 @@
+{{- $db := index .Values "timescaledb-single" -}}
+{{- if and $db.enabled (eq .Values.promscale.connectionSecretName "") (eq .Values.promscale.connection.password "") -}}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Name }}-promscale-initializer-sa
+  namespace: {{ template "tobs.namespace" . }}
+  labels:
+    app: {{ template "tobs.fullname" . }}
+    chart: {{ template "tobs.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .Release.Name }}-promscale-initializer-role
+  namespace: {{ template "tobs.namespace" . }}
+  labels:
+    app: {{ template "tobs.fullname" . }}
+    chart: {{ template "tobs.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "patch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ .Release.Name }}-promscale-initializer-rolebinding
+  namespace: {{ template "tobs.namespace" . }}
+  labels:
+    app: {{ template "tobs.fullname" . }}
+    chart: {{ template "tobs.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ .Release.Name }}-promscale-initializer-role
+subjects:
+- namespace: {{ template "tobs.namespace" . }}
+  kind: ServiceAccount
+  name: {{ .Release.Name }}-promscale-initializer-sa
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-connection-initializer
+  namespace: {{ template "tobs.namespace" . }}
+  labels:
+    app: {{ template "tobs.fullname" . }}
+    chart: {{ template "tobs.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  password-initializer.sh: |
+    #!/bin/bash
+    while ! kubectl get secret {{ .Release.Name }}-credentials --namespace {{ template "tobs.namespace" . }}; do
+      echo "Waiting for {{ .Release.Name }}-credentials secret."
+      sleep 1
+    done
+    PASS=$(kubectl get secret --namespace {{ template "tobs.namespace" . }} {{ .Release.Name }}-credentials -o json | jq -r '.data["PATRONI_SUPERUSER_PASSWORD"]')
+    kubectl get secret --namespace {{ template "tobs.namespace" . }} {{ .Release.Name }}-promscale -o json | jq --arg PASS "$PASS" '.data["PROMSCALE_DB_PASSWORD"]=$PASS' | kubectl apply -f -
+---
+apiVersion: batch/v1
+kind: Job
```

Comment:
Should this job run on both upgrades and fresh installs? It seems like it runs on both but I am not sure that we want it in upgrade paths?

Reply:
I don't see a way it would break the installation. However since this is needed only during the first install, it might be better to run it once.

Good call! I changed it to run only once in latest commit.